pkg/storage/storage.go

```diff
@@ -29,4 +29,7 @@ type Backend interface {
 
 	// ClearDeleted is used clear marked delete records.
 	ClearDeleted(ctx context.Context, cutoff time.Time)
+
+	// Sync is used to notify about changes happen in storage.
+	Sync(ctx context.Context) chan struct{}
```

Comment:
I guess the intention here is to use the context Done as a signal to stop syncing? Otherwise there's no way to communicate that syncing should stop. This could be error prone to use.

Reply:
@calebdoxsey See https://github.com/pomerium/pomerium/pull/1135#discussion_r460088546
 
`ctx.Done()` will trigger the `close(ch)` inside `Sync`, which will signal the caller to stop syncing.

Reply:
I updated the method name, and the documentation, hope it's more clear now.